### ISSUE TITLE
Some ligatures still enabled in regex groups

### ISSUE THREAD

**Issue by chimericdream** [tonsky/FiraCode#847]
I notied this when reading through the changelog for v2 (AWESOME update by the way, thank you!). A few of the ligatures listed as being disabled are still being partly applied. I copied the text and made a quick gif to show which ones are still behaving a little oddly.

This is probably an edge case (I don't think I would have ever noticed if not for the fact that I have Fira Code enabled in a GitHub user style), but thought you'd like to know.

![ligatures](https://user-images.githubusercontent.com/198075/65269957-cc337f80-dadf-11e9-946e-5bd465cf8369.gif)

**Comment by tonsky** [comment]
That is actually how I inteded this to work. The part that belongs to regex is isolated. Everything that comes after it is collapased as standalone ligatures. If regex is `(?=>>` then it’s `(?=` (lookahead) followed by `>>`

**Comment by chimericdream** [comment]
In that case, shouldn't a few more of the ligatures be working? I'm pretty decent at regex, but I don't use some of these operators. So it's entirely possible I am missing some fine detail in it. But if your goal is to isolate the lookaround part of the string, shouldn't lines 3, 9, 10, 16, and 20 above behave like the others that are changing?

Admittedly, font design is not my forte, so let me know if I am off base. I saw something that seemed like it might have been a small oversight, but I'd be lying if I said I was 100% sure.

**Comment by tonsky** [comment]
You are right, they should! The reason why they are not working is different: I disable ligatures by default if they are immediately preceded or followed by the same char. E.g. no `=>` in `==>` or `=>>`. Initially it was added to isolate two-char and three char ligatures better (in case there is `==>` ligature it won’t trigger `=>` one first. This is a tricky problem and this is the best I could do at the moment. At some point in the future I might also revisit that decision if I find a way to solve it better. Thanks for bringing my attention to that!
